### ISSUE TITLE
Encountering "Not enough tokens on the account" Error During Minting in Metaplex

### ISSUE THREAD

**Issue by Beor18** [metaplex-foundation/metaplex#2305]
**Steps to Reproduce:**

- Set up my Candy Machine on Metaplex.
- Attempted to perform a mint.
- Encountered the error "Not enough tokens on the account".

**Environment:**

- Metaplex Javascript API version used: 4.12.0
- Metaplex JavaScript SDK version used: 0.20.1
- Blockchain network (Mainnet, Devnet, etc.): devnet

**Attempted Solutions:**

- Checked my account balance to confirm sufficient funds.
- Reviewed Candy Machine configuration for potential errors.

**Question to the Community:**

Has anyone else experienced this issue? I am interested in knowing if it's a common error and if there are any known solutions or recommended diagnostic steps.

Any help or guidance on how to resolve this issue would be greatly appreciated. I am also willing to provide more information if needed for diagnosing this issue.

Thanks you!

**Comment by xatxay** [comment]
Hey, did you find the solution to this? i'm encoutering this error as well

**Comment by Beor18** [comment]
> Hey, did you find the solution to this? i'm encoutering this error as well


Yes, you have to migrate the JS SDK library to the UMI Framework, which is highly recommended and in the mint you are not missing any configuration such as parameters

**Comment by SkyYap** [comment]
Same. I have almost 1.5 sol yet fail to mint. 

```transaction 1 failed: NotEnoughTokens: Not enough tokens on the account

Source: Program > mplCandyGuard [Guard1JwRhJkVH6XZhzoYxeBVQe872VH6QggF4BWmS9g]

Caused By: Error: failed to send transaction: Transaction simulation failed: Error processing Instruction 2: custom program error: 0x1784```
